features/subscription_attach_restrictions.feature

```diff
@@ -0,0 +1,130 @@
+    Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+    When I set the machine token overlay to the following yaml:
+      """
+      machineTokenInfo:
+          contractInfo:
+              resourceEntitlements:
+                  - type: support
+                    affordances:
+                        onlySeries: <onlyseries>
+      """
+    When I attempt to attach `contract_token` with sudo
+    Then stderr contains substring:
+      """
+      Attaching to this contract is only allowed on the Ubuntu <onlyseries>
+      """
+    And the machine is unattached
+
+    Examples: ubuntu release
+      | release | machine_type  | onlyseries |
+      | xenial  | lxd-container | bionic     |
+      | bionic  | lxd-container | xenial     |
+      | focal   | lxd-container | noble      |
+      | jammy   | lxd-container | focal      |
+      | noble   | lxd-container | jammy      |
+
+  Scenario Outline: Check notice visible if attach fail when restricted to release
+    Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+    When I set the machine token overlay to the following yaml:
+      """
+      machineTokenInfo:
+          contractInfo:
+              resourceEntitlements:
+                  - type: support
+                    affordances:
+                        onlySeries: <onlyseries>
+      """
+    When I attempt to attach `contract_token` with sudo
+    Then stderr contains substring:
+      """
+      Attaching to this contract is only allowed on the Ubuntu <onlyseries>
+      """
+    And the machine is unattached
+    When I run `pro status` with sudo
+    Then stdout contains substring:
+      """
+      Limited to release: Ubuntu <onlyseries>
+      """
+
+    Examples: ubuntu release
+      | release | machine_type  | onlyseries |
+      | xenial  | lxd-container | bionic     |
+      | bionic  | lxd-container | xenial     |
+      | focal   | lxd-container | noble      |
+      | jammy   | lxd-container | focal      |
+      | noble   | lxd-container | jammy      |
+
+  Scenario Outline: Attach Successfully if current release same as restricted release
+    Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+    When I set the machine token overlay to the following yaml:
+      """
+      machineTokenInfo:
+          contractInfo:
+              resourceEntitlements:
+                  - type: support
+                    affordances:
+                        onlySeries: <onlyseries>
+      """
+    When I attempt to attach `contract_token` with sudo
+    Then the machine is attached
+
+    Examples: ubuntu release
+      | release | machine_type  | onlyseries |
+      | xenial  | lxd-container | xenial     |
+      | bionic  | lxd-container | bionic     |
+      | focal   | lxd-container | focal      |
+      | jammy   | lxd-container | jammy      |
+      | noble   | lxd-container | noble      |
```

Comment:
The test right below this one is a superset of this test so we can delete this one.